lib/ansible/module_utils/vmware.py

```diff
@@ -234,11 +234,11 @@ def compile_folder_path_for_object(vobj):
     thisobj = vobj
     while hasattr(thisobj, 'parent'):
         thisobj = thisobj.parent
+        if thisobj.name == 'Datacenters':
```

Comment:
Yes. 'Datacenters' is parent of all object so I don't think so there will be anything above it. If you see code is doing `paths.reverse()` so `Datacenters` will be the last candidate we are looking for. Can anyone confirm this assumption?